tests/libostreetest.c

```diff
@@ -72,29 +72,44 @@ ot_test_setup_repo (GCancellable *cancellable,
   return g_steal_pointer (&ret_repo);
 }
 
+gboolean
+ot_check_for_overlay (gboolean  *on_overlay,
+                      GError   **error)
+{
+  /* Keep this in sync with the overlayfs bits in libtest.sh */
+  struct statfs stbuf;
+  const char *pwd = g_getenv ("PWD");
```

Comment:
Surely `g_get_current_dir()` (needs to be assigned to a `g_autofree gchar *`), or simply replace `pwd` with `"."` below?

I'm not completely sure of the startup sequence of this stuff, but if the test isn't guaranteed to have been started by a shell designed for interactive use, then I don't think `PWD` is guaranteed to be up-to-date.

According to its documentation, `g_get_current_dir()` returns `PWD`, but only after verifying that `PWD` is in fact the same directory as `"."`.

Reply:
Sure. I had started with `getenv ("TEST_TMPDIR")` at first, though changed it afterwards since it doesn't actually appear when running with gdtr.